cli_tools/gce_onestep_image_import/onestep_importer/utils.go

```diff
@@ -0,0 +1,163 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package importer
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/param"
+)
+
+// runCmd executes the named program with given arguments.
+// Logs the running command.
+func runCmd(name string, args []string) error {
+	log.Printf("Running command: '%s %s'", name, strings.Join(args, " "))
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+// runCmdAndGetOutput executes the named program with given arguments.
+// Logs the running command. Returns the output of the executed program, and error if any.
+func runCmdAndGetOutput(cmdString string, args []string) ([]byte, error) {
+	log.Printf("Running command: '%s %s'", cmdString, strings.Join(args, " "))
+	return runCmdAndGetOutputWithoutLog(cmdString, args)
+}
+
+// runCmdAndGetOutputWithoutLog executes the named program with given arguments.
+// Returns the output of the executed program, and error if any.
+func runCmdAndGetOutputWithoutLog(cmdString string, args []string) ([]byte, error) {
+	output, err := exec.Command(cmdString, args...).CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("%v%v", string(output), err)
+	}
+	return output, nil
+}
+
+// importFromCloudProvider imports image from the specified cloud provider
+func importFromCloudProvider(args *OneStepImportArguments) error {
+	var exportedGCSPath string
+
+	// 1. import from specified cloud provider
```

Comment:
put it in a separate function, calling "newImporterForCloudProvider", then importer.run